### DISCUSSION TITLE
Image building performance vs Docker

### DISCUSSION THREAD

**Discussion by suptejas** [containers/podman#19087] [Q&A]
Hi, I was doing some research about Podman, and [came across the following few sentences](https://www.solutelabs.com/blog/podman-vs-docker):

> Podman provides faster startup thanks to its daemon-less structure. At any point in time, the host system stays protected. However, when it comes to overall building, Docker is a winner. The building part relies primarily on image building. Since Docker has native image building, it can build at least six times faster than Podman.

I am interested in the building part - is the above statement still true and are there any plans to improve build times by using a native image building system if this hasn't already been built out?

**Comment by flouthoc** [comment]
Hi @suptejas , I am not sure if I can totally agree with the statement above, both `docker` and `podman` allows building images in different configuration like with different storage drivers, rootless vs rootful etc. I am not at all sure if there is any official bench-marking which has been done to compare build performance if different settings and environments.

However I do agree that there are some design differences between `podman` and `docker` which are intentional for the use-case which podman provides, other than that I am not sure if there is any official bench-marking which has been done between `buildah` and `buildkit` which is used by `podman` and `docker` respectively. @containers/podman-maintainers Might be able to confirm more about if any such comparisons were done in past.

> build times by using a native image building system

Podman uses buildah natively. I am not sure what do you mean by `using a native image building system`.

**Reply by vrothberg** [reply]
"at least six times faster" feels extremely generalized.  I am unaware of that rule of thumb and assume the authors used fuse-overlayfs and did not compare root vs root and rootless vs rootless.  In such a scenario, I would not expect serious performance differences with either tool.  buildkit, however, seems to have some great performance boosts in certain scenarios.

**Comment by afbjorklund** [comment]
Docker has changed their "native building" to use BuildKit instead. Which _improved_ the performance.

https://docs.docker.com/engine/release-notes/23.0/#2300

> Set Buildx and BuildKit as the default builder on Linux. moby/moby # 43992
> * Alias docker build to docker buildx build.  docker/cli # 3314

**Comment by rhatdan** [comment]
The only way I see that they could improve performance would be with caching of previous builds or in handling of parallel entries in the Containerfile. @flouthoc has been adding a lot of these caching mechanisms to Buildah. If we had specific workloads that showed the difference in speed, then we could work to improve Buildah caching.

@nalind any other ideas?
